liblz4: Update to 1.9.1 and switch to codeload

neheb opened the pull request:
Simplifies the Makefile.

Remove build hacks and use PKG_INSTALL.

Use PKG_BUILD_PARALLEL for faster compilation.

Switch to the Compression submenu to group this with the others.

Added lz4 binaries to be installable.

Signed-off-by: Rosen Penev <rosenp@gmail.com>

Maintainer: @dajhorn 
Compile tested: arc700

BKPepe commented on libs/liblz4/Makefile:
```diff
@@ -7,54 +7,75 @@
-	  CFLAGS="$(TARGET_CFLAGS)" \
-	  PREFIX="$(CONFIGURE_PREFIX)" \
-	  install
+define Package/zstd/description
```
Isn't this wrong?

neheb replied:
Yes it is. Copy/Paste error

neheb commented:
I think the package should be renamed to lz4 and moved to the utils directory. Thoughts?

cotequeiroz commented:
If the main use is as a library, let it be. OpenSSL also has an utility. There are plenty of options that point the other way as well. I don’t have a firm opinion, but if the utility has just been added, it is an indication that the library is the main component.

neheb commented:
Fixed typo in commit message.

@cotequeiroz It would allow removal of the PKG_BUILD_DIR line. I also see that zstd is under utils (Both were written by the same person).

cotequeiroz commented:
I don’t really have even a preference. If you think it is the best, go right ahead!

jonathanunderwood commented:
lz4 primary use case is as a library. I'd recommend leaving it where it is.

diizzyy commented:
The change itself looks good however we should probably evaluate ```-O3``` (upstream seems to prefer O3 looking at the Makefile) and possibly enabling LTO.

neheb commented:
Added O3 optimization option based on zlib's solution.

Switched to the included CMake build. This has a few advantages:

Elimination of InstallDev

lz4 is linked with liblz4.so. With the default Makefile it's statically linked, increasing the filesize massively. I couldn't find a way to modify the Makefile to link to the .so.

diizzyy commented:
LGTM

cotequeiroz commented on libs/liblz4/Makefile:
```diff
@@ -7,54 +7,80 @@
+PKG_BUILD_DIR:=$(BUILD_DIR)/lz4-$(PKG_VERSION)
+
 PKG_MAINTAINER:=Darik Horn <dajhorn@vanadac.com>
+PKG_LICENSE:=BSD-2-Clause GPL-2.0-only
```
I don't know how pedantic we are going to be with regards of license information.   The library is licensed under the BSD-2-Clause, and the app is GPL-2.0 or later (look at the preamble in files, such as `programs/lz4cli.c`, including the "any later version" text.

IMO, we can use `LICENSE:=BSD-2-Clause`; `LICENSE_FILES:=LICENSE lib/LICENSE` under `Package/liblz4`, and `LICENSE=GPL-2.0-or-later`; `LICENSE_FILES:=LICENSE programs/COPYING` under `Package/lz4`.

neheb replied:
I like this. edited.

cotequeiroz commented on libs/liblz4/Config.in:
```diff
@@ -0,0 +1,11 @@
+menu "Configuration"
```
Add the option to the library, not to the util, otherwise the option will only be visible under the library package if you select the utility.
Instead of opening a new menu, which adds one line to the Compression menu, use that line for the option itself instead, avoiding the need to open a new menu, by removing the `menu` line and adding the `depends` to the `config` definition:
```
config LZ4_OPTIMIZE_SPEED
	bool "Optimize for speed"
	depends on PACKAGE_liblz4
	help
		This enables additional optimization and
		increases performance considerably at
		the expense of binary size.

endif
```
will turn
```
<*> liblz4........................................ Extremely fast compression
      Configuration  --->
```
into
```
<*> liblz4........................................ Extremely fast compression
[ ]   Optimize for speed (NEW)
```
If you look at the current menu, notice that libzstd's option is not done right.

neheb replied:
I like this as well. Fixed.

I will probably fix libzstd when I get the chance.

cotequeiroz commented on libs/liblz4/Makefile:
```diff
@@ -7,54 +7,83 @@
+
 define Package/liblz4/install
 	$(INSTALL_DIR) $(1)/usr/lib
 	$(CP) $(PKG_INSTALL_DIR)/usr/lib/liblz4.so* $(1)/usr/lib/
```
```suggestion
 	$(CP) $(PKG_INSTALL_DIR)/usr/lib/liblz4.so.$(ABI_VERSION)* $(1)/usr/lib/
```
to help catching a bad `ABI_VERSION`, or for simplicity:
```suggestion
 	$(CP) $(PKG_INSTALL_DIR)/usr/lib/liblz4.so.* $(1)/usr/lib/
```
We should not copy the unversioned `.so` file, to allow multiple ABI-versions of the lib to coexist.

neheb replied:
Meh OK. I doubt a situation like that will happen but I see no harm in doing so.

cotequeiroz commented on libs/liblz4/Makefile:
```diff
@@ -7,54 +7,83 @@
+  TITLE:=Extremely fast compression
+  LICENSE:=BSD-2-Clause
+  LICENSE_FILES:=LICENSE lib/LICENSE
+  MENU:=1
```
```
  ABI_VERSION:=$(firstword $(subst .,$(space),$(PKG_VERSION)))
```
ABI version seems to be stable.

neheb commented on libs/liblz4/Makefile:
```diff
@@ -7,54 +7,83 @@
-PKG_SOURCE_URL:=https://github.com/lz4/lz4.git
-PKG_SOURCE_VERSION:=$(PKG_VERSION)
+PKG_BUILD_PARALLEL:=1
+CMAAKE_INSTALL:=1
```
Interestingly enough, it still runs Build/InstallDev. If you notice in the CMAKE_OPTIONS, I have the mandir set to /dev/null because of some weird race condition that made every second build fail.

cotequeiroz commented on libs/liblz4/Makefile:
```diff
@@ -7,54 +7,84 @@
+  SECTION:=utils
+  CATEGORY:=Utilities
+  TITLE:=Extremely fast compression
+  LICENSE=GPL-2.0-or-later
```
```suggestion
  LICENSE:=GPL-2.0-or-later
```
